src/nvim/mapping.c

```diff
@@ -2810,6 +2811,25 @@ void modify_keymap(uint64_t channel_id, Buffer buffer, bool is_unmap, String mod
   xfree(parsed_args.desc);
 }
 
+/// abbrmap_array: Retrieves abbreviation mappings for a specified mode and buffer.
+///
+/// @param int_mode The integer representation of the mapping mode (e.g., insert, command).
+/// @param buf      Pointer to the buffer for which the mappings are to be retrieved.
+///                 If NULL, global mappings are retrieved.
+/// @param mappings Pointer to an array where the retrieved mappings will be stored.
+static void abbrmap_array(int int_mode, buf_T *buf, Array *mappings)
```

Comment:
I don't think this should be a separate function. It should use the same loop as mappings, just with different loop conditions.

Reply:
emm currently empty string also works not correct right ?  it should return all items which include  abbr mode map also. i think ..